### PR TITLE
KNJ-11054 make the command expire all purposes

### PR DESCRIPTION
**Pull request by cedel1**

### CODE REVIEW

File: test_requirements.txt [hunk 1]
```diff
@@ -2,6 +2,7 @@ flake8==3.8.4
 mypy==0.790
 python-dateutil==2.7.5
 django-extensions
-freezegun==0.3.11
+freezegun==0.3.12
```
Comment (cedel1) [inline test_requirements.txt:5]: 0.3.11 is not compatible with python 3.8 - `time.clock()` function has been removed.

File: test_requirements.txt [hunk 2]
```diff
@@ -2,6 +2,7 @@ flake8==3.8.4
...
 Faker==1.0.1
 django-reversion==3.0.8
+django-germanium
```
Comment (cedel1) [inline test_requirements.txt:8]: need this for `test_call_command` function.

File: test_requirements.txt (hunk 1 above)
Comment (cedel1) [reply test_requirements.txt:5]: (same version as our core, even though freezegun is at 1.1.0 right now)

File: tests/tests/test_deactivate_expired_reasons.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        self.assertTrue(LegalReason.objects.exists_deactivated_consent(MARKETING_SLUG, self.customer))
+
+    def test_command_should_not_anonymize_customer_if_expiring_reason_is_not_related_to_customer_date(self):
+        marketing_legal_reason = LegalReason.objects.create_consent(MARKETING_SLUG, self.customer)
+        marketing_legal_reason.save()
+        LegalReason.objects.create_consent(FIRST_AND_LAST_NAME_SLUG, self.customer).save()
+
+        # none of the consents have expired
+        test_call_command('deactivate_expired_reasons')
+        self.assertTrue(LegalReason.objects.exists_valid_consent(MARKETING_SLUG, self.customer))
```
Comment (matllubos) [inline tests/tests/test_deactivate_expired_reasons.py]: you installed germanium. Now you can use assert_true

**Review by Formulka — APPROVED** [review]

File: test_requirements.txt (hunk 2 above)
Comment (rubickcz) [reply test_requirements.txt:8]: You could make instance of that command and call it manually, or? Nevermind.

File: tests/tests/test_deactivate_expired_reasons.py [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+        # consent has not expired
+        test_call_command('deactivate_expired_reasons')
+        assert_true(LegalReason.objects.exists_deactivated_consent(MARKETING_SLUG, self.customer))
+
+        # consent has expired
+        with freeze_time(legal_reason.expires_at + relativedelta(seconds=1)):
+            test_call_command('deactivate_expired_reasons')
+        assert_true(LegalReason.objects.exists_deactivated_consent(MARKETING_SLUG, self.customer))
+
+    def test_command_should_not_anonymize_customer_if_expiring_reason_is_not_related_to_customer_date(self):
```
Comment (rubickcz) [inline tests/tests/test_deactivate_expired_reasons.py]: date -> data?

**Review by rubickcz — APPROVED** [review]

File: tests/tests/test_deactivate_expired_reasons.py [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+        # consent has not expired
+        test_call_command('deactivate_expired_reasons')
+        assert_true(LegalReason.objects.exists_valid_consent(MARKETING_SLUG, self.customer))
+
+        # consent has expired
+        with freeze_time(legal_reason.expires_at + relativedelta(seconds=1)):
+            test_call_command('deactivate_expired_reasons')
+        assert_false(LegalReason.objects.exists_valid_consent(MARKETING_SLUG, self.customer))
+
+    def test_command_should_not_expire_deactivated_legal_reasons(self):
```
Comment (rubickcz) [inline tests/tests/test_deactivate_expired_reasons.py:34]: Maybe also test, that the command will not touch already expired consents (i.e, those with EXPIRED state).

**Review by rubickcz — CHANGES_REQUESTED** [review]
I would add one more test and pls fix the typo - see PR comments.

File: tests/tests/test_deactivate_expired_reasons.py (hunk 4 above)
Comment (cedel1) [reply tests/tests/test_deactivate_expired_reasons.py]: Corrected.

File: tests/tests/test_deactivate_expired_reasons.py (hunk 5 above)
Comment (cedel1) [reply tests/tests/test_deactivate_expired_reasons.py:34]: Added.

**Pull request merged**